ceremonies/src/benchmarking.rs

```diff
@@ -0,0 +1,386 @@
+{
+	pair.public().into()
+}
+
+fn prove_attendance<T: Config>(
+	prover: T::AccountId,
+	cid: CommunityIdentifier,
+	cindex: CeremonyIndexType,
+	attendee: &sr25519::Pair,
+) -> ProofOfAttendance<T::Signature, T::AccountId>
+where
+	<T as frame_system::Config>::AccountId: From<sp_core::sr25519::Public>,
+	<T as Config>::Signature: From<sp_core::sr25519::Signature>,
+{
+	let msg = (prover.clone(), cindex);
+	ProofOfAttendance {
+		prover_public: prover,
+		community_identifier: cid,
+		ceremony_index: cindex,
+		attendee_public: account_id::<T>(&attendee),
+		attendee_signature: T::Signature::from(attendee.sign(&msg.encode())),
+	}
+}
+
+fn get_all_claims<T: Config>(
+	attestees: &Vec<sr25519::Pair>,
+	cid: CommunityIdentifier,
+	cindex: CeremonyIndexType,
+	mindex: MeetupIndexType,
+	location: Location,
+	timestamp: T::Moment,
+	n_participants: u32,
+) -> Vec<ClaimOfAttendance<T::Signature, T::AccountId, T::Moment>>
+where
+	<T as frame_system::Config>::AccountId: From<sp_core::sr25519::Public>,
+	<T as Config>::Signature: From<sp_core::sr25519::Signature>,
+{
+	let mut claims: Vec<ClaimOfAttendance<T::Signature, T::AccountId, T::Moment>> = vec![];
+	for a in attestees {
+		claims.push(
+			ClaimOfAttendance::<T::Signature, T::AccountId, T::Moment>::new_unsigned(
+				a.public().into(),
+				cindex,
+				cid,
+				mindex,
+				location,
+				timestamp,
+				n_participants,
+			)
+			.sign(a),
+		);
+	}
+	claims
+}
+
+/// Run until a particular block.
+fn run_to_block<T: Config>(n: T::BlockNumber)
+where
+	<T as frame_system::Config>::BlockNumber: From<u64>,
+	<T as pallet_timestamp::Config>::Moment: From<u64>,
+{
+	while frame_system::Pallet::<T>::block_number() < n {
+		if frame_system::Pallet::<T>::block_number() > 1.into() {
+			frame_system::Pallet::<T>::on_finalize(frame_system::Pallet::<T>::block_number());
+		}
+		let new_timestamp: u64 = (T::BlockNumber::from(GENESIS_TIME) +
+			T::BlockNumber::from(BLOCKTIME) * n)
+			.unique_saturated_into();
+
+		set_timestamp::<T>(T::Moment::from(new_timestamp));
+		pallet_timestamp::Pallet::<T>::on_finalize(frame_system::Pallet::<T>::block_number());
+		frame_system::Pallet::<T>::set_block_number(
+			frame_system::Pallet::<T>::block_number() + 1.into(),
+		);
+		frame_system::Pallet::<T>::on_initialize(frame_system::Pallet::<T>::block_number());
+	}
+}
+
+/// Progress blocks until the phase changes
+fn run_to_next_phase<T: Config>()
+where
+	<T as frame_system::Config>::BlockNumber: From<u64>,
+	<T as pallet_timestamp::Config>::Moment: From<u64>,
+{
+	let phase = encointer_scheduler::Pallet::<T>::current_phase();
+	let mut blocknr = frame_system::Pallet::<T>::block_number();
+	while phase == encointer_scheduler::Pallet::<T>::current_phase() {
+		blocknr += 1.into();
+		run_to_block::<T>(blocknr);
+	}
+}
+
+pub fn set_timestamp<T: Config>(t: T::Moment) {
+	let _ = pallet_timestamp::Pallet::<T>::set(RawOrigin::None.into(), t);
+}
+
+fn get_proof_of_attendance<T: Config>(
```

Comment:
I know this is copied code, but the naming confuses me.

* `get_proof_of_attendance`: I'd expect to only get the proof, not actually inserting the fake reputation into the runtime
* `prove_attendance`: This one I'd rather expect to change state in the runtime instead of just returning the proof object.

Please change:
* rename: `get_proof_of_attendance` -> `fake_last_attendance_and_get_proof`
* rename: `prove_attendance`: `create_proof_of_attendance`

I happily accept other suggestions.